in_winlog: add a new plugin for Windows Event Log

fujimotos opened the pull request:
This adds basic support for Windows Event Log. This plugin can:

 1. Read and parse Event Log records using Win32 API.
 2. Subscribe to multiple log channels at once.
 3. Preserve the read offset between restart (using SQLite3)

The biggest limitation right now is that it uses non-Unicode functions.
I'm planning to add Unicode support in the subsequent PRs.

Signed-off-by: Fujimoto Seiji <fujimoto@clear-code.com>

fujimotos commented:
Documentation: fluent/fluent-bit-docs/pull/176
Related issue: #960

edsiper commented on plugins/in_winlog/in_winlog.c:
```diff
@@ -0,0 +1,353 @@
+    /* Initialize context */
+    ctx = flb_calloc(1, sizeof(struct flb_in_winlog_config));
+    if (!ctx) {
+        perror("malloc");
```
please use flb_errno()

fujimotos replied:
I fixed it to use `flb_errno()`.

edsiper commented on plugins/in_winlog/in_winlog.c:
```diff
@@ -0,0 +1,353 @@
+    ctx->bufsize = DEFAULT_BUFFER_SIZE;
+    ctx->buf = flb_malloc(ctx->bufsize);
+    if (!ctx->buf) {
+        perror("malloc");
```
flb_errno();

fujimotos replied:
I fixed it to use `flb_errno()` too.

edsiper commented on plugins/in_winlog/winlog.h:
```diff
@@ -0,0 +1,95 @@
+    unsigned int record_number;
+    unsigned int time_written;
+    unsigned int seek;
+    struct winlog_channel *next;
```
Please do not use old-fashion linked list, use struct mk_list instead

fujimotos replied:
Now it uses `struct mk_list` to manage multiple channels.

fujimotos commented:
Thank you for the code review! I'll follow them up next week.

fujimotos commented:
@edsiper I've fixed the code according to your suggestion.

The testing failure on AppVeyor is due to the recent changes introduced
into master, which I addressed in the following commits: 

* https://github.com/edsiper/chunkio/pull/42
* https://github.com/edsiper/chunkio/pull/44
* https://github.com/fluent/fluent-bit/pull/1395

With all of them merged, I can confirm that AppVeyor runs green again

edsiper commented:
thanks
